lib/normalize.js

```diff
@@ -67,31 +67,59 @@ const normalizeSources = dp => {
   for (const idx in dp.sources) {
     dp.sources[idx].title = dp.sources[idx].name
   }
-  for (const idx in dp.resources){
-    for (const id in dp.resources[idx].sources){
-      if(dp.resources[idx].sources[id].title) {
+  for (const idx in dp.resources) {
+    for (const id in dp.resources[idx].sources) {
+      if (dp.resources[idx].sources[id].title) {
         dp.resources[idx].sources[id].title = dp.resources[idx].sources[id].title
       }
-      else if(dp.resources[idx].sources[id].name){
+      else if (dp.resources[idx].sources[id].name) {
         dp.resources[idx].sources[id].title = dp.resources[idx].sources[id].name
       }
       else {
-        dp.resources[idx].sources[id].title = 'unkown'
+        dp.resources[idx].sources[id].title = 'unknown'
       }
     }
   }
   return dp
 }
 
 const normalizeContributors = dp => {
-  for (const idx in dp.contributors){
-    dp.contributors[idx].name = dp.contributors[idx].name.toLowerCase().replace(/\s/g, '_')
+  const normalizeContirutorsTitle = dp => {
+    for (const idx in dp.contributors) {
+      if (dp.contributors[idx].title) {
+        dp.contributors[idx].title = dp.contributors[idx].title
+      } else if (dp.contributors[idx].name) {
+        dp.contributors[idx].title = dp.contributors[idx].name
+        delete dp.contributors[idx].name
+      }
+    }
+  }
+  if (dp.author) {
+    dp.contributors.push({title: dp.author})
```

Comment:
This would fail if you don't have ` contributors` property saying that it cannot find method push of undefined. So you need to: 

- check if contributors exists
  - yes - check if it is an array (or are we already handling it somewhere?)
    - yes - continue
    - no - make it array
  - no - create it as an empty array and continue

Reply:
@anuveyatsu i will do it, this `check if it is an array` is handled by another function